draft-issuance.md

```diff
@@ -0,0 +1,169 @@
+subsidy size over time will remain the same and the funds for block subsidies will last
+a similar amount of time. Instead, it focuses solely on enhancing the predictability
+and consistency of the block subsidy issuance process.
+
+Smoothing the emissions curve helps ensure that the network remains economically
+viable and stable as it transitions from a traditional issuance mechanism to one
+that maintains a sustainable and predictable issuance of rewards over time. It
+prevents abrupt changes in the rate of newly issued coins, which could lead to
+disruptions in the network's economic model and potentially impact its security
+and sustainability. A smoother emissions curve allows for a more gradual and controlled
+transition, providing ZEC stakeholders and participants with a clear understanding of
+how rewards will be distributed over time.
+
+
+
+# Specification
+
+Smoothing the issuance curve is possible using an exponential decay formula that
+satisfies the following requirements:
+
+## Requirements
+
+1. Block subsidies MUST be weakly decreasing
+2. Block subsidies SHOULD approximate a continuous function
+3. When `AVAILABLE_SUBSIDIES(h) > 0` then block subsidies for block `h`
+MUST always be `> 0`, preventing a final “unmined” zatoshi
+4. For any 4 year period, all paid out block subsidies MUST equal approximately
+half of `AVAILABLE_SUBSIDIES` at the beginning of that 4 year period
+5. This functionality MUST be introduced as part of a network upgrade
+
+The above requirements assume no deflationary action, i.e. that no ZEC is added
+to `AVAILABLE_SUBSIDIES`. They are referenced below as **Rn**.
+
+## Solution
+
+Given the block height `h` define a function **S**, such that:
+
+**S(h)** = Block subsidy for a given `h`, that satisfies above requirements.
+
+Please note that
+
+`AVAILABLE_SUBSIDIES(h+1) = AVAILABLE_SUBSIDIES(h) - S(h)` assuming no deflationary action.
+
+An exponential decay function **S** satisfies **R1** and **R2** above:
+
+`S(h) = BLOCK_SUBSIDY_FRACTION * AVAILABLE_SUBSIDIES(h)`
+
+Finally, to satisfy **R3** above we need to always round up to at least 1 Zatoshi
+if `AVAILABLE_SUBSIDIES(h) > 0`:
+
+`S(h) = ROUND_UP(BLOCK_SUBSIDY_FRACTION * AVAILABLE_SUBSIDIES(h))`
+
+# Rationale
+
+## `BLOCK_SUBSIDY_FRACTION`
+
+That value of `41 / 100_000_000` was selected so that it satisfies the equation:
+
+`(1 - BLOCK_SUBSIDY_FRACTION)^NUMBER_OF_BLOCKS_IN_4_YEARS ~ ½`
+
+Meaning after a period of 4 years around half of `AVAILABLE_SUBSIDIES` will be paid out
+as block subsidies, thus satisfying **R4**.
+
+
+## Other Notes
+
+The suggested implementation avoids using float numbers. Rust and C++ will both round
+the result of the final division up, satisfying **R3** above.
+
+# Appendix: Simulation
+
+We encourage readers to run the following Rust code, which simulates block subsidies.
+According to this simulation, assuming no deflationary action, block subsidies would
+last for approximately 113 years:
```

Comment:
Nitpick: It is 113.9 years from August 2023.

[This Rust code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f8f50d0883b20b677b43f62be49f2377) gives me the final lines:
```sh
$ cargo run --release | tail 
   Compiling zec-smooth-issuance v0.1.0 (/Users/mail/zec-smooth-issuance)
    Finished release [optimized] target(s) in 0.80s
     Running `target/release/zec-smooth-issuance`
47921512 (113.9 years): 1(1.0e-8 ZEC) 9(9.0e-8 ZEC)
47921513 (113.9 years): 1(1.0e-8 ZEC) 8(8.0e-8 ZEC)
47921514 (113.9 years): 1(1.0e-8 ZEC) 7(7.0e-8 ZEC)
47921515 (113.9 years): 1(1.0e-8 ZEC) 6(6.0e-8 ZEC)
47921516 (113.9 years): 1(1.0e-8 ZEC) 5(5.0e-8 ZEC)
47921517 (113.9 years): 1(1.0e-8 ZEC) 4(4.0e-8 ZEC)
47921518 (113.9 years): 1(1.0e-8 ZEC) 3(3.0e-8 ZEC)
47921519 (113.9 years): 1(1.0e-8 ZEC) 2(2.0e-8 ZEC)
47921520 (113.9 years): 1(1.0e-8 ZEC) 1(1.0e-8 ZEC)
47921521 (113.9 years): 1(1.0e-8 ZEC) 0(0.0e0 ZEC)
```

```suggestion
last for approximately 114 years from August 2023:
```